README.md
# Project Description

An example-only UneeQ frontend UI project built using uneeq-js SDK and React. This codebase serves as an example for the developers of UneeQ’s partners / customers to build their own frontend project.

# Project delivered as-is

There is no support express or implied for this repository. While we take care to ensure compatibility and update dependencies from time to time for security reasons, you are encouraged to do your own due diligence on the dependencies used with this project. Neither P22 nor UneeQ are responsible for any issues or trouble, damages or conflicts arising from your use of this project. Please see [LICENCE.md](./LICENCE.md) for full licence.

Please see [LICENCE.md](./LICENCE.md) for full licence.

# Packages

## uneeq-react-core

Provides React state management for uneeq-js-sdk

## uneeq-react-ui

Provides a generic, customisable user interface

## uneeq-download-pdf

Renders PDFs of a conversation transcript or user-saved information for download

# Getting Started

## Create your own copy of this repo

[Fork on Gitlab](fork)

**or**

Clone [uneeq-oss/react-frontend](repo) as "upstream"

> `git clone --origin upstream git@gitlab.com:uneeq-oss/react-frontend.git`

Add your own remote as "origin"

> `git remote add origin git@github.com:you/your_repo.git`

## Create your project

Create a copy of the directory `example-app`.

> `cp example-app my-app`

Edit `package.json`

- Add your new app to `workspaces`
- Add `dev` and `build` scripts
- Edit the default `build` and `start` scripts _(optional)_

e.g.

```JSON
...
"workspaces": [
  "packages/*",
  "example-app",
  "my-app"
],
"scripts": {
  "start-my-app-only": "yarn workspace my-app start",
  "start-my-app": "concurrently \"yarn:start-packages\" \"yarn:start-my-app-only\"",
  "build-my-app-only": "yarn workspace my-app build",
  "build-my-app": "yarn build-packages && yarn build-example-only",

  "start": "yarn start-my-app",
  "build": "yarn build-my-app",

  ...
}
...
```

## Configure

Rename `my-app/.env.example` to `.env`, add your UneeQ details

## Install Dependencies

Run `yarn install`

## Run Locally

Run `yarn start`

## Customize

Edit `my-app/src/config.js`

## Build for Production

Run `yarn build`  
Copy `my-app/build` to your webserver

## Update

Avoid editing code withing `packages`. This will allow you to update your fork from [react-frontend](repo) in future.

Read the CHANGELOG to understand any changes that have been made, especially breaking changes.

Update your local:

<!-- TODO: Should we suggest merging a specific version by tag?-->

```sh
git fetch upstream
git merge upstream/master
```

See [Syncing a fork](fork-sync) for more.

# Integration

## Commands

Commands can be sent to control the front-end UI. They must be send as stringified JSON within instructions->displayHtml->html

```json
{
  "instructions": {
    "displayHtml": {
      "html": "{\"openInputBar\":true}"
    }
  }
}
```

```js
const command = { openInputBar: true };
const x = {
  instructions: {
    displayHtml: {
      html: JSON.stringify(command),
    },
  },
};
```

## Available Commands

### Keyboard input bar trigger

Switch to text input mode so the user can type a response. Useful in combination with a response like "Sorry, I didn't understand you. Could you please type that?". `true` indicates switching to text input mode, `false` indicates switching back to speach input.

**Command**: `{"openInputBar": true}`

### Chat window trigger

Show the transcript window. `true` indicates showing the users transcript, `false` indicates hiding/closing the transcript.

**Command**: `{"openTranscript": true}`

### Suggested responses

The content of suggested responses will only be triggered from the NLP platform. The suggested responses have a title and each has a label to be displayed on screen along with an utterance to be send when the user clicks that response.

**Command**:

```JSON
{
 	"mainTitle": "Call to action text: ",
 	"suggestedResponses": [{
 		"label": "Suggested Response #1",
 		"utterance": "response 1"
 	}, {
 		"label": "Suggested Response #2",
 		"utterance": "response 2"
 	}, {
 		"label": "Suggested Response #3",
 		"utterance": "response 3"
 	}]
 }
```

### On-screen information trigger

The information content can only be triggered from the NLP platform. This is used to provide users with useful snippets of information.

#### Video

**Command**: `{ "information": [{ "type": "video", "source": "https://www.youtube.com/embed/rF2u7RTPsHI"}] }`

Full example (including heading):

```json
{
  "instructions": {
    "displayHtml": {
      "html": "{\"information\": [{\"type\": \"heading\", \"text\": \"Video heading here\"},{ \"type\": \"video\", \"source\": \"https://www.youtube.com/embed/rF2u7RTPsHI\", \"width\": \"100%\", \"height\": \"375\"}]}"
    }
  }
}
```

##### Information "video" Type Properties

These properties can be changed to adjust the sizing and appearance of the video.
Additionally the source value can be changed to suit any embed URL of services
where the original embed code is an iframe (e.g. Vimeo)

The video information type has not been tested with videos hosted on Amazon S3 or similar services.

| property | value                                     | type            | required |
| -------- | ----------------------------------------- | --------------- | -------- |
| type     | video                                     | string          | yes      |
| source   | https://www.youtube.com/embed/xxxxxxx     | string          | yes      |
| width    | 100% (default) or numeric value           | string / number | no       |
| height   | 375 (default) or numeric/percentage value | string / number | no       |

#### Markdown

**Command**: `{ "information": "#Heading\nInformaion\n[link](http://example.com)" }`

See [On-screen information markdown](#On-screen-information-markdown) below for more

### Feedback form trigger

Ability to trigger the feedback form from within the conversation. `true` indicates to show the feedback dialog.

**Command**: `{"openFeedback": true}`

### Human escalation form trigger

Ability to trigger the human escalation form from within the conversation which includes prompting the user for email address (if email address have not been previously provided). `true` indicates showing the escalation form.

**Command**: `{"openEscalationForm": true}`

# On-screen information markdown

[Markdown Cheatsheet](markdown-cheatsheet)

## Currently Supported tokens

- Headings: `# Heading`
- Bold: `**bold**`
- Italic: `*italic*`
- Lists: `- list item`
- Links: `[link](http://example.com)`
- Images: `![alt text](http://example.com/icon.png "Title")`

## Clickable suggested responses in markdown

URLs can be in format `say:UTTERANCE` where `UTTERANCE` is the word or phrase that should be automatically spoken to the digital human when the link is clicked.

E.g.

```
[link](say:URL encoded utterance)
```

Full example:

```
{ "information": "#Choose An Option: \n\n[Accept the deal](say:Yes I would like the deal)" }
```

# Update policy

- Semver
- Fixes may not be ported to older versions

# Test

TODO

## testState

Allows the welcome & initialisation to be skipped to go directly to a certain state

- Base [testState=%7B%7D](http://localhost:3000/?testState=%7B%7D)
- Loading [testState=%7B%22ready%22%3Afalse%2C%22loadingPercentage%22%3A40%7D](http://localhost:3000/?testState=%7B%22ready%22%3Afalse%2C%22loadingPercentage%22%3A40%7D)
- Busy [testState=%7B%22unavailable%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22unavailable%22%3Atrue%7D)
- Declined [testState=%7B%22permissionAllowed%22%3Afalse%7D](http://localhost:3000/?testState=%7B%22permissionAllowed%22%3Afalse%7D)
- Transcript [testState=%7B%22transcriptOpen%22%3Atrue%2C%22transcript%22%3A%5B%7B%22message%22%3A%22Does%20this%20look%20right%3F%22%2C%22time%22%3A%221970-01-01T00%3A00%3A00.000Z%22%2C%22user%22%3Afalse%7D%2C%7B%22message%22%3A%22Yes%22%2C%22time%22%3A%221970-01-01T00%3A00%3A00.000Z%22%2C%22user%22%3Atrue%7D%5D%7D](http://localhost:3000/?testState=%7B%22transcriptOpen%22%3Atrue%2C%22transcript%22%3A%5B%7B%22message%22%3A%22Does%20this%20look%20right%3F%22%2C%22time%22%3A%221970-01-01T00%3A00%3A00.000Z%22%2C%22user%22%3Afalse%7D%2C%7B%22message%22%3A%22Yes%22%2C%22time%22%3A%221970-01-01T00%3A00%3A00.000Z%22%2C%22user%22%3Atrue%7D%5D%7D)
- Transcript Minimised [testState=%7B%22transcriptHasOpened%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22transcriptHasOpened%22%3Atrue%7D)
- User Question [testState=%7B%22question%22%3A%22Does%20this%20look%20right%3F%22%7D](http://localhost:3000/?testState=%7B%22question%22%3A%22Does%20this%20look%20right%3F%22%7D)
- With Saved Items [testState=%7B%22savedItems%22%3A%5B0%5D%7D](http://localhost:3000/?testState=%7B%22savedItems%22%3A%5B0%5D%7D)
- Input Error [testState=%7B%22noInput%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22noInput%22%3Atrue%7D)
- With Saved Items and Minimised Transcript [testState=%7B%22savedItems%22%3A%5B0%5D%2C%22transcriptHasOpened%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22savedItems%22%3A%5B0%5D%2C%22transcriptHasOpened%22%3Atrue%7D)
- Text Input [testState=%7B%22inputMode%22%3A%22text%22%7D](http://localhost:3000/?testState=%7B%22inputMode%22%3A%22text%22%7D)
- End Session [testState=%7B%22endConfirmOpen%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22endConfirmOpen%22%3Atrue%7D)
- Escalation Form [testState=%7B%22escalationFormOpen%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22escalationFormOpen%22%3Atrue%7D)
- Feedback [testState=%7B%22feedbackOpen%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22feedbackOpen%22%3Atrue%7D)
- Settings [testState=%7B%22settingsOpen%22%3Atrue%2C%22selectedDevices%22%3A%7B%7D%2C%22devices%22%3A%7B%22audioInput%22%3A%5B%5D%2C%22videoInput%22%3A%5B%5D%2C%22audioOutput%22%3A%5B%5D%7D%7D](http://localhost:3000/?testState=%7B%22settingsOpen%22%3Atrue%2C%22selectedDevices%22%3A%7B%7D%2C%22devices%22%3A%7B%22audioInput%22%3A%5B%5D%2C%22videoInput%22%3A%5B%5D%2C%22audioOutput%22%3A%5B%5D%7D%7D)
- Menu [testState=%7B%22menuOpen%22%3Atrue%7D](http://localhost:3000/?testState=%7B%22menuOpen%22%3Atrue%7D)
- On-screen Info [testState=%7B%22onScreenInfo%22%3A%7B%22information%22%3A%5B%7B%22type%22%3A%22markdown%22%2C%22markdown%22%3A%22%23%20Heading%5CnParagraph%5Cn\*%20Bullet%20%5Blink%5D(%2F)%22%7D%5D%7D%7D](<http://localhost:3000/?testState=%7B%22onScreenInfo%22%3A%7B%22information%22%3A%5B%7B%22type%22%3A%22markdown%22%2C%22markdown%22%3A%22%23%20Heading%5CnParagraph%5Cn*%20Bullet%20%5Blink%5D(%2F)%22%7D%5D%7D%7D>)
- Expanded On-screen Info [testState=%7B%22onScreenInfo%22%3A%7B%22information%22%3A%5B%7B%22type%22%3A%22markdown%22%2C%22markdown%22%3A%22%23%20Heading%5CnParagraph%5Cn\*%20Bullet%20%5Blink%5D(%2F)%22%7D%5D%7D%2C%22expandedInfo%22%3Atrue%7D](<http://localhost:3000/?testState=%7B%22onScreenInfo%22%3A%7B%22information%22%3A%5B%7B%22type%22%3A%22markdown%22%2C%22markdown%22%3A%22%23%20Heading%5CnParagraph%5Cn*%20Bullet%20%5Blink%5D(%2F)%22%7D%5D%7D%2C%22expandedInfo%22%3Atrue%7D>)
- Suggested Responses [testState=%7B%22onScreenInfo%22%3A%7B%22suggestedResponses%22%3A%7B%22id%22%3A1%2C%22mainTitle%22%3A%22You%20can%20ask%20me%20about%3A%22%2C%22suggestedResponses%22%3A%5B%7B%22label%22%3A%22Suggestion%201%22%2C%22utterance%22%3A%221%22%7D%2C%7B%22label%22%3A%22Suggestion%202%22%2C%22utterance%22%3A%222%22%7D%2C%7B%22label%22%3A%22Suggestion%203%22%2C%22utterance%22%3A%223%22%7D%5D%7D%7D%7D\* ](http://localhost:3000/?testState=%7B%22onScreenInfo%22%3A%7B%22suggestedResponses%22%3A%7B%22id%22%3A1%2C%22mainTitle%22%3A%22You%20can%20ask%20me%20about%3A%22%2C%22suggestedResponses%22%3A%5B%7B%22label%22%3A%22Suggestion%201%22%2C%22utterance%22%3A%221%22%7D%2C%7B%22label%22%3A%22Suggestion%202%22%2C%22utterance%22%3A%222%22%7D%2C%7B%22label%22%3A%22Suggestion%203%22%2C%22utterance%22%3A%223%22%7D%5D%7D%7D%7D*)

# Licence

No warranty or support is implied by UneeQ, please refer to LICENCE.md

[repo]: https://gitlab.com/uneeq-oss/react-frontend
[fork]: https://docs.gitlab.com/ee/user/project/repository/forking_workflow.html#creating-a-fork
[fork-sync]: https://docs.gitlab.com/ee/user/project/repository/forking_workflow.html
[markdown-cheatsheet]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet


CHANGELOG.md
# Changelog

## [1.3.0] - 2020-12-17

### Changed

- Added video type to `InformationContent` component.

E.g:

```json
{
  "instructions": {
    "displayHtml": {
      "html": "{\"information\": [{\"type\": \"heading\", \"text\": \"Video heading here\"},{ \"type\": \"video\", \"source\": \"https://www.youtube.com/embed/rF2u7RTPsHI\"}]}"
    }
  }
}
```

## [1.2.0] - 2020-12-08

### Changed

- Markdown in `InformationContent` component now accepts URLs in format `say:UTTERANCE` where `UTTERANCE` is the word or phrase that should be automatically spoken to the digital human when the link is clicked.

E.g.

```json
[Accept the deal](say:Yes I would like the deal)
```

### Fixed

- Added extra checks to uneeqCommandReducer for suggestedResponses in case an empty array of suggested responses is sent from backend conversation service.

## [1.1.2] - 2020-09-23

### Changed

- Allow sendLocalAudio:false

## [1.1.1] - 2020-09-17

### Fixed

- Call onTimedOut at end of timeout
- Center Suggested Responses on tablet-size layout
- Adjust bottom limit of OnScreenInfo to avoid overlap with user question text

### Changed

- Add 'video' to example privacy policy

## [1.1.0] - 2020-09-11

### Changed

- Colors for Push-to-Talk button are now set in the theme
  - see example-app/src/theme.ts `// PTT animation` and `// PTT desktop colors & gradients`
- PDF Downloads can include SessionId
  - see example-app/src/app/downloadPdf.ts
- Feedback form remembers user input
- Improvements to example-app home page

### Fixed

- Bug causing Escalation Form to show when `config.showEscalationForm=false`
- PDF download bug on Android
- Passcode Overlay scroll bug

## [1.0.0] - 2020-09-02

### Added

- Generic analytics implementation - no longer reliant on Google Analytics (see `config.analytics`)
- Add `postinstall` script to build packages
- Increased test coverage

### Changed

- Update `uneeq-js` to `0.30`
- Improve onscreen-information layout
- rename CONVERSATION_ID to PERSONA_ID in example-app
  - If you were to overwrite your own `config.ts` with `example-app/src/config.ts` you would also need to update your `.env` file to specify `REACT_APP_UNEEQ_PERSONA_ID`
- make privacy policy a blank template

### Docs

- Improve avatarPosition docs in config.js

### Fixed

- Layout bugs in Home and Passcode
- Saved items state not updating

### Removed

- Google Analytics - Generic analytics implementation added

### Tidy

- Merge & deduplicate OnScreenInfo translations
- Update packages and readme to reflect new repo location
- License field in package.json files
- Use to generic avatar name "Digital Human"


example-app/src/theme.ts
const colors = {
  text: '#000000',
  textLight: '#393939',
  textAlternate: '#FFFFFF',

  primary: '#225bb5',
  primaryLight: '#4287f5',
  primaryText: '#FFFFFF',

  secondary: '#22b5a1',
  secondaryLight: '#4bd6c4',
  secondaryText: '#FFFFFF',

  greyDark: '#50575D',
  grey: '#BDC0C7',
  greyLight: '#F6F6F6',

  error: '#BF2828',
  warning: '#FBBD1D',

  // TODO create our own progress bar with rebass so we can use variant="timoutBar"
  timeoutColor: '#1C8728',
  timeoutTrailColor: '#BDC0C7'
}

const hex2rgba = (hex: string, alpha = 1) => {
  const [r, g, b] = hex.match(/\w\w/g)!.map(x => parseInt(x, 16))
  return `rgba(${r},${g},${b},${alpha})`
}

export const theme = {
  colors: {
    ...colors,

    // PTT animation
    pttMobileGlow: hex2rgba(colors.primaryLight, 0.2), // Mobile - glow around button while recording
    // PTT desktop colors & gradients
    pttColRecording: 'white',
    pttImgRecording: `radial-gradient(circle, ${colors.primaryLight} 0%, white 100%)`,
    pttColSending: 'primaryLight',
    pttImgSending: `radial-gradient(circle, white 0%,${colors.primaryLight} 100%)`,
    pttColWaiting: 'primaryLight',
    pttImgWaiting: 'primary'
  },
  fonts: {
    body: 'Overpass, Roboto, system-ui, sans-serif',
    heading: 'inherit'
  },
  fontSizes: ['14px', '15px', '16px', '20px', '22px', '24px', '30px'],
  fontWeights: {
    body: 400,
    heading: 400,
    bold: 700
  },
  lineHeights: {
    body: 1.5,
    heading: '31px'
  },
  space: [
    0,
    4,
    8,
    10,
    12,
    14,
    16,
    18,
    20,
    22,
    24,
    28,
    32,
    40,
    48,
    56,
    64,
    80,
    100,
    128
  ],
  radii: {
    button: 2,
    card: 8
  },
  shadows: {
    card: '0 4px 4px rgba(0, 0, 0, 0.25)'
  },
  text: {
    heading: {
      fontFamily: 'heading',
      lineHeight: 'heading',
      fontWeight: 'heading',
      fontSize: 5,
      pb: 2,
      mt: 3,
      mx: 4
    }
  },
  variants: {
    link: {
      color: 'primary'
    },
    card: {
      bg: 'white',
      color: 'textLight',
      borderRadius: 'card',
      p: [6, 6, 12, 12, 12, 12],
      boxShadow: 'card'
    },
    modal: {
      variant: 'variants.card',
      label: 'modal',
      width: ['92%', '92%', 850],
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center'
    },
    transcript: {
      borderRadius: '3px',
      overflow: 'hidden',
      top: 0,
      right: 0,
      width: ['100%', '100%', '100%', '100%', 238, 238],
      height: ['100%', '100%', '100%', '100%', 504, 504],
      zIndex: 3,
      color: 'text',
      backgroundColor: 'greyLight',
      '& .transcript-header': {
        height: 62,
        position: ['fixed', 'fixed', 'fixed', 'fixed', 'static'],
        backgroundColor: 'primary',
        color: 'text',
        top: [0, 0, 0, 0, 'auto', 'auto'],
        zIndex: 10
      },
      '& .transcript-content': {
        // borderTop: '1px solid #E6E6E6',
        borderBottom: '1px solid #E6E6E6',
        fontSize: [4, 1],
        p: 3,
        height: ['100%', '100%', '100%', '100%', 'auto', 'auto'],
        paddingTop: [62, 62, 62, 62, 3, 3],
        paddingBottom: [53, 53, 53, 53, 3, 3]
      },
      '& .transcript-footer': {
        height: 45,
        position: ['fixed', 'fixed', 'fixed', 'fixed', 'static'],
        minHeight: '45px',
        bottom: [0, 0, 0, 0, 'auto', 'auto'],
        backgroundColor: 'white',
        width: '100%',
        p: 2
      }
    }
  },
  buttons: {
    // TODO
    unstyled: {
      background: 'transparent',
      color: 'secondary',
      fontFamily: 'inherit',
      fontSize: 0
    },
    primary: {
      fontFamily: 'inherit',
      borderRadius: 'button',
      fontWeight: '600',
      fontSize: '16px',
      lineHeight: '24px',
      verticalPadding: '12px',
      horizontalPadding: '22px',
      transition: 'all 1s',
      cursor: 'pointer',
      backgroundColor: 'primary',
      border: '1px solid transparent',
      borderColor: 'primary',
      whiteSpace: 'nowrap',
      '& svg, & img': {
        fill: 'currentColor'
      },
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      color: 'text',
      '& img, & svg': {
        mr: 2
      },
      '&:hover': {
        backgroundColor: 'primaryLight'
      },
      '&:focus': {
        backgroundColor: 'primary',
        '--box-shadow-color': 'primary',
        boxShadow: '0 0 3px var(--box-shadow-color)',
        borderColor: 'primaryLight'
      },
      '&:disabled': {
        backgroundColor: 'greyDark',
        color: '#85898E'
      }
    },
    primaryInverted: {
      variant: 'buttons.primary',
      bg: 'black',
      color: 'primary',
      borderColor: 'primary',
      '&:hover': {
        backgroundColor: 'rgba(1, 99, 172, 0.1)'
      },
      '&:focus': {
        outline: 'none',
        '--box-shadow-color': 'primary',
        boxShadow: '0 0 3px var(--box-shadow-color)'
      },
      '&:disabled': {
        borderColor: 'grey',
        color: '#E3E5E8'
      }
    },
    secondary: {
      variant: 'buttons.primary',
      color: 'secondaryText',
      backgroundColor: 'secondary',
      borderColor: 'secondary',
      '& svg, & img': {
        // TODO
        // fill: 'currentColor'
        fill: 'red'
      },
      '& img, & svg': {
        mr: 2
      },
      '&:hover': {
        backgroundColor: 'secondaryLight'
      },
      '&:focus': {
        backgroundColor: 'secondary',
        '--box-shadow-color': 'secondary',
        boxShadow: '0 0 3px var(--box-shadow-color)',
        borderColor: 'secondaryLight'
      },
      '&:disabled': {
        backgroundColor: 'greyDark',
        color: '#85898E'
      }
    },
    secondaryInverted: {
      variant: 'buttons.secondary',
      color: 'secondary',
      background: 'white',
      borderColor: 'secondary',
      '& svg, & img': {
        // fill: 'currentColor'
        fill: 'red'
      },
      '& img, & svg': {
        // mr: 2
        mr: 20
        // TODO
      },
      '&:hover': {
        backgroundColor: '#2E2818'
      },
      '&:focus': {
        borderColor: 'secondary',
        '--box-shadow-color': 'secondary',
        boxShadow: '0 0 3px var(--box-shadow-color)'
      },
      '&:disabled': {
        borderColor: 'greyDark',
        color: '#85898E'
      }
    },
    outline: {
      color: 'textAlternate',
      fontWeight: '300',
      cursor: 'pointer',
      border: '1px solid #FFFFFF',
      transition: 'all 1s',
      backgroundColor: 'transparent',
      borderRadius: 50,
      '&:hover': {
        backgroundColor: 'rgba(255,255,255,0.1)'
      },
      '&:focus': {
        outline: 'none',
        border: '1px solid transparent',
        borderColor: 'white',
        '--box-shadow-color': 'transparent',
        boxShadow: 'none'
      },
      '&:disabled': {
        backgroundColor: 'thirdGray',
        color: 'fourthGray'
      }
    }
  },
  breakpoints: ['320px', '600px', '768px', '1024px', '1280px']
}

export default theme


example-app/src/config.ts
import downloadPdf from './app/downloadPdf'

let environment = {}
// You can override default settings for different environments
// by setting REACT_APP_CONFIG environment variable
if (process.env.REACT_APP_CONFIG === 'EXAMPLE') {
  console.info('config: example')
  environment = {
    conversationId: 'OTHER'
  }
} else {
  console.info('config: default')
}

const config = {
  debug: true,

  url: process.env.REACT_APP_UNEEQ_URL!,
  conversationId: process.env.REACT_APP_UNEEQ_PERSONA_ID!,
  tokenUrl: process.env.REACT_APP_UNEEQ_TOKEN_URL!,

  // Is a passcode needed to obtain a token? Show a passcode screen so the user can provide it.
  usePasscode: false,
  // Recaptcha - if used with passcode
  recaptchaSiteKey: process.env.REACT_APP_RECAPTCHA_SITE_KEY!,

  // welcome can be suppressed during dev with REACT_APP_NO_WELCOME=1
  playWelcome: !process.env.REACT_APP_NO_WELCOME,

  sendLocalVideo: false,
  customData: {},

  // When holding down spacebar, durations shorter that this will trigger
  // a message like "Keep holding spacebar while you talk"
  tapThreshold: 700, // ms

  // How long can the user be inactive before timeout
  timeout: 480 * 1000, // ms
  // How close to the end of the timeout should we show the warning
  timeoutWarning: 180 * 1000, // ms
  // e.g. timeout=90sec, timeoutWarning=30sec - after 60 secs of inactivity warning will show, after 90 secs sessions ends

  // If the avatar is not centred within the video avatarPosition needs to be changed.
  // This will ensure correct positioning narrow (mobile) screens
  avatarAspect: 16 / 9, // video aspect ratio
  avatarPosition: 0.5, // Where is the avatar in the video frame (0.5 = center, 0.33 = 1/3 from the left)

  informationInTranscript: false,

  // show we show the contact form after Feedback?
  showEscalationForm: false,

  // How many empty transcripts before an error is shown
  emptyTranscriptThreshold: 3,
  avatarName: 'a Digital Human',

  downloadPdf,

  // Optional - track events
  // analytics: (label: string, action?: string) => {
  //   console.info('analytics event:', { label, action })
  //   // Send the event to your analytics backend
  //   // Example Google Universal Analytics (gtag.js)
  //   // https://developers.google.com/analytics/devguides/collection/gtagjs
  //   gtag('event', 'action', {
  //     event_category: 'my-category',
  //     event_label: label
  //   })
  // },

  ...environment
}

export default config


example-app/src/app/Home/styles.ts
export const styles = {
  mainContainer: {
    '&::-webkit-scrollbar, & *::-webkit-scrollbar': {
      display: 'none'
    },
    '& *': {
      msOverflowStyle: 'none',
      scrollbarWidth: 'none'
    },
    msOverflowStyle: 'none',
    scrollbarWidth: 'none',
    label: 'homeMainContainer',
    minWidth: '100%',
    height: '100%',
    position: 'absolute',
    alignItems: 'flex-start',
    flexDirection: ['column', 'column', 'column', 'column', 'row', 'row'],
    justifyContent: [
      'flex-start',
      'flex-start',
      'flex-start',
      'flex-start',
      'normal',
      'normal'
    ],
    width: '100%',
    '& > video': {
      objectFit: 'cover',
      width: '100vw',
      height: [
        'calc(100vh - 48px)',
        'calc(100vh - 48px)',
        '100vh',
        '100vh',
        '100vh'
      ],
      minHeight: ['auto', 'auto', '100vh', '100vh', '100vh'],
      position: ['static', 'static', 'static', 'static', 'fixed', 'fixed'],
      top: 0,
      left: 0,
      zIndex: 0
    },
    maxHeight: '100%'
  },
  videoContainer: {
    my: 2,
    width: '28%',
    display: ['none', 'none', 'none', 'none', 'block', 'block']
  },
  textContainer: {
    label: 'text-container',
    mb: [4, 0],
    pl: ['10%', '10%', '10%', '10%', '15%', '15%'],
    pr: ['10%', '10%', '10%', '10%', '40%', '35%'],
    mt: 0,
    height: ['1px', '1px', '1px', '1px', 'auto', 'auto'],
    maxWidth: ['100%', '100%', '100%', '100%', '100%', '90%'],
    color: 'textAlternate',
    textAlign: 'left',
    zIndex: 4,
    width: '100%',
    maxHeight: '100%',
    overflowY: ['visible', 'visible', 'visible', 'visible', 'scroll', 'scroll']
  },
  mobileImSophie: {
    display: ['flex', 'flex', 'flex', 'flex', 'none', 'none'],
    fontSize: [2, 2, 6, 6, 78, 78],
    lineHeight: '26px',
    minHeight: 56,
    textAlign: 'center',
    color: 'white',
    fontWeight: 'bold',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center'
  },
  imSophieText: {
    display: ['none', 'none', 'none', 'none', 'flex', 'flex'],
    fontSize: 78,
    lineHeight: '80px',
    mt: 15,
    fontWeight: 'bold'
  },
  subtitle: {
    fontWeight: '300',
    fontSize: 4,
    mt: 2,
    mb: [3, 4],
    lineHeight: '30px',
    display: ['none', 'none', 'none', 'none', 'block', 'block'],
    maxWidth: 480
  },
  sophieBGImage: {
    label: 'sophieBGImage',
    position: 'fixed',
    backgroundSize: ['100%', '100%', '70%'],
    backgroundRepeat: 'no-repeat',
    backgroundPosition: ['top', 'top', '100% bottom'],
    top: ['3em', '3em', 'unset'],
    right: ['3em', '3em', 'unset'], // this is bc the empty space the img has, we should ask them for a better png
    height: '100%',
    width: '100%',
    zIndex: 1
  },
  letsChatButton: {
    lineHeight: '24px',
    fontWeight: '600',
    minWidth: '10rem',
    py: [3, 2],
    px: [6, 5],
    mt: 3,
    mb: 10
  },
  privacyTextToggle: {
    fontWeight: '300',
    cursor: 'pointer',
    fontSize: 0,
    lineHeight: '22px',
    textDecoration: 'none',
    maxWidth: 480
  },
  disclaimer: {
    mt: 9,
    mb: 6,
    fontWeight: '300',
    fontSize: 0,
    lineHeight: '22px',
    '& a, & a:active': {
      color: 'primary',
      textDecoration: 'none'
    },
    '& a:hover': {
      textDecoration: 'underline'
    },
    maxWidth: 480
  },
  recaptchaDisclaimer: {
    fontWeight: '300',
    fontSize: 0,
    lineHeight: '22px',
    mb: 14
  },
  startButton: {
    label: 'startButton',
    display: ['block', 'block', 'block', 'block', 'none', 'none'],
    mt: [-100, -100, -200, -200, 1, 1],
    mb: [100, 100, 150, 150, 1, 1],
    mx: ['auto', 'auto', 'auto', 'auto', 0, 0],
    zIndex: 4
  },
  surveyToggle: {
    fontWeight: 'bold',
    cursor: 'pointer',
    fontSize: 2,
    lineHeight: '24px',
    textDecoration: 'none',
    mb: 2,
    mt: [10, 10, 10, 10, 0, 0],
    pt: [10, 10, 10, 10, 0, 0]
  },
  togglePrivacyIcon: {
    display: 'inline-block',
    color: 'primary'
  },
  surveyContainer: {
    transition: 'max-height .5s linear',
    overflow: 'hidden',
    '& label': {
      fontWeight: '300',
      fontSize: 2,
      alignItems: 'flex-start',
      lineHeight: '24px'
    },
    '& textarea': {
      borderRadius: 'button',
      borderColor: 'grey',
      backgroundColor: 'white'
    },
    maxWidth: 440
  },
  checkbox: {
    width: '20px',
    height: '20px',
    borderRadius: 'button',
    'input:checked ~ &': {
      color: 'text',
      backgroundColor: 'white'
    }
  },
  textarea: {
    color: 'text',
    fontFamily: 'sans-serif',
    fontSize: 1
  },
  paragraphContainer: {
    transition: 'max-height .5s linear',
    overflow: 'hidden',
    fontSize: '12px',
    lineHeight: '18px',
    flexDirection: 'column',
    fontWeight: '300',
    px: 0,
    mt: 2,
    '& p': {
      mb: 3
    },
    maxWidth: 480
  },
  mobileParagraphsContainer: {
    zIndex: 9,
    color: 'white',
    textAlign: 'center',
    position: 'absolute',
    left: 0,
    bottom: 0,
    width: '100%',
    height: '100%',
    flexDirection: 'column',
    backgroundColor: 'rgba(0,0,0,0.9)',
    justifyContent: 'center',
    alignItems: 'center',
    fontWeight: '300',
    fontSize: 4,
    px: 7,
    my: 2,
    '& p': {
      my: 3
    }
  },
  privacyPolicyLink: {
    textDecoration: 'underline',
    color: 'currentColor'
  },
  titleContainer: {
    display: 'flex',
    width: '100%'
  }
}
export default styles
